### DISCUSSION TITLE
How to use typer to parse a string inside a program

### DISCUSSION THREAD

**Discussion by Lucas256-cmd** [fastapi/typer#736] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer


@app.command()
def hello(name: str):
    """Say hello to NAME."""
    typer.echo(f"Hello {name}")

@app.command()
def goodbye(name: str):
    """Say hello to NAME."""
    typer.echo(f"Hello {name}")


print("""This program can say hello or goodbye. To say hello to someone type hello + the name of the person
To say goodbye to someone type goodbye + the name of the person""")
command = input("What do you want to do? ")

# now let typer handle the input in command. I don't know how to do this, but I think something like this:
typer.run(command)  # I read the docs, so I know this won't work, but I want typer to handle the input as if it was run from the command line.
```

### Description

Basically I want to make some kind bash shell in python with typer to parse the commands, provide help pages, etc. In the code you can see an example of what I mean. I know you can pass a command with CliRunner but that gives problems with promt() and then I still have to do some parsing myself. I think I found something similar [here](https://github.com/tiangolo/typer/issues/106) but I don't think this is what I want. If it is then I definitely don't know how to implement it :)

So my question is if this (processing strings and executing and everything) is possible with typer. 
If not do you know a module or something else that I can use? 

Thanks in advance

### Operating System

Windows

### Typer Version

0.9.0

### Python Version

3.11.0
